File: spring-project/src/main/java/com/example/spring_project/entities/Employee.java
package com.example.spring_project.entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "employees")
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "full_name", nullable = false, length = 160)
    private String fullName;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(name = "phone_number", length = 10)
    private String phoneNumber;

    private Boolean active;

    @Column(name = "hashed_password")
    private String hashedPassword;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: reactjs-project/src/types/Employee.ts
export interface Employee {
  id: number;
  fullName: string;
  email: string;
  dateOfBirth: string;
  gender: 'NAM' | 'NU' | 'KHAC';
  phoneNumber: string;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEmployeeRequest {
  fullName: string;
  email: string;
  dateOfBirth: string;
  gender: 'NAM' | 'NU' | 'KHAC';
  phoneNumber: string;
  active: boolean;
  password: string;
}

export interface UpdateEmployeeRequest {
  fullName: string;
  dateOfBirth: string;
  gender: 'NAM' | 'NU' | 'KHAC';
  phoneNumber: string;
  active: boolean;
  password: string;
}
